src/epcpm/sunspectocsv.py

```diff
@@ -187,28 +193,27 @@ def gen(self) -> int:
         """
         self.wrapped.children[0].check_offsets_and_length()
 
-        non_header_block_length = sum(
-            child.check_offsets_and_length() for child in self.wrapped.children[1:]
-        )
-
-        # Per SunSpec model specification, pad with a 16-bit pad to force even alignment to 32-bit boundaries.
-        add_padding = (non_header_block_length % 2) == 1
-        if add_padding:
-            non_header_block_length += 1
-
         # Track the total accumulated length of the model's registers.
         accumulated_length = 0
-
         rows = []
 
         model_types = ["Header", "Fixed Block", "Repeating Block"]
-
         zipped = zip(enumerate(self.wrapped.children), model_types)
         for (model_type_index, child), model_type in zipped:
-            # TODO: Check if this is one of the spots where "add_padding" is incorrect for tables.
+            pre_pad_block_length = child.check_offsets_and_length()
+            # Per SunSpec model specification, pad with a 16-bit pad to force even alignment to 32-bit boundaries.
+            add_padding = (pre_pad_block_length % 2) == 1
```

Comment:
modulus against a magic num

Reply:
This is a standard python methodology to check for odd integer. I could make it a method, but I think it's overkill. Is the comment not explicit enough?

Reply:
ah yeah you're right, disregard